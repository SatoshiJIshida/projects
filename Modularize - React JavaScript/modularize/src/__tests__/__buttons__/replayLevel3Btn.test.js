import React from "react";
import { shallow } from "enzyme";
import { render, screen, fireEvent } from "@testing-library/react";
import ReplayLevel3Btn from "../../components/buttons/replayLevel3Btn";

/**
 * Library Jest taken from source https://jestjs.io/
 * Library Enzyme taken from source https://www.npmjs.com/package/enzyme
 * Library React Testing Library taken from source https://testing-library.com/docs/react-testing-library/intro/
 */

describe("Button that opens Level3 (repeated button with different text))", () => {
  it("should be defined", () => {
    expect(ReplayLevel3Btn).toBeDefined();
  });

  it("should render correctly", () => {
    const button = shallow(<ReplayLevel3Btn name="button-test" />);
    expect(button).toMatchSnapshot();
  });

  it("should call mock function when clicked", () => {
    const handleClick = jest.fn();
    render(<ReplayLevel3Btn onClick={handleClick} />);
    fireEvent.click(screen.getByText(/Replay Level 3/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
