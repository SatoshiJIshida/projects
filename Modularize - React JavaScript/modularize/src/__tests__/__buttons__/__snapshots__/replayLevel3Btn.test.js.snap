// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button that opens Level3 (repeated button with different text)) should render correctly 1`] = `ShallowWrapper {}`;
